gensrc/thrift/Descriptors.thrift

```diff
@@ -402,6 +402,9 @@ struct TIcebergTable {
 
     // partition column names
     4: optional list<string> partition_column_names
+
+    // Map from partition id to partition metadata.
+    5: optional map<i64, THdfsPartition> partitions
```

Comment:
I don't see where we use it

Reply:
we send it in icebergtable toThrift, and use it in IcebergTableDescriptor